ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -565,13 +565,18 @@ forkBlockForging IS{..} blockForging =
         trace $ TraceNodeIsLeader currentSlot
 
         -- Tick the ledger state for the 'SlotNo' we're producing a block for
-        let tickedLedgerState :: Ticked (LedgerState blk)
-            tickedLedgerState =
+        tickedLedgerState :: Ticked (LedgerState blk) <-
+            lift . evaluate $ -- TODO is evaluate enough here?
```

Comment:
Please leave the `let` as it was and just call `_ <- evaluate tickedLedgerState` immediately prior to `trace`.